### PR TITLE
Nb/nlp/tweet use case

### PR DESCRIPTION
**Pull request by Nadav-Barak**
Added tweet emotion dataset + tests based on it to different checks

### CODE REVIEW

File: deepchecks/nlp/datasets/classification/tweet_emotion.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        Otherwise, returns a single object.
+
+    Returns
+    -------
+    dataset : Union[TextData, pd.DataFrame]
+        the data object, corresponding to the data_format attribute.
+    train, test : Tuple[Union[TextData, pd.DataFrame],Union[TextData, pd.DataFrame]
+        tuple if as_train_test = True. Tuple of two objects represents the dataset splitted to train and test sets.
+    """
+    if data_format not in ['TextData', 'Dataframe']:
```
Comment (nirhutnik) [inline deepchecks/nlp/datasets/classification/tweet_emotion.py:58]:
```suggestion
    if data_format.lower() not in ['textdata', 'dataframe']:
```

File: deepchecks/nlp/datasets/classification/tweet_emotion.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        if data_format == 'TextData':
+            dataset = TextData(dataset.text, label=dataset[_target], task_type='text_classification',
+                               additional_data=dataset.drop(columns=[_target, 'text']))
+        return dataset
+    else:
+        # train has more sport and Customer Complains but less Terror and Optimism
+        train = dataset[dataset['train_test_split'] == 'Train'].drop(columns=['train_test_split'])
+        test = dataset[dataset['train_test_split'] == 'Test'].drop(columns=['train_test_split'])
+
+        if data_format == 'TextData':
```
Comment (nirhutnik) [inline deepchecks/nlp/datasets/classification/tweet_emotion.py:73]:
```suggestion
        if data_format.lower() == 'textdata':
```

File: deepchecks/nlp/datasets/classification/tweet_emotion.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+__all__ = ['load_data', 'load_embeddings', 'load_precalculated_predictions']
+
+_FULL_DATA_URL = 'https://ndownloader.figshare.com/files/39265559'
+_EMBEDDINGS_URL = 'https://ndownloader.figshare.com/files/39264332'
+_PREDICTIONS_URL = 'https://ndownloader.figshare.com/files/39264461'
+_target = 'label'
+
+
+def load_data(data_format: str = 'TextData', as_train_test: bool = True) -> \
```
Comment (nirhutnik) [inline deepchecks/nlp/datasets/classification/tweet_emotion.py:34]: Save the data locally and then load it from local if already exists? That's what we usually do

File: deepchecks/nlp/text_data.py [hunk 4]
```diff
@@ -254,7 +256,7 @@ def sample(self: TDataset, n_samples: int, replace: bool = False, random_state:
             samples = samples[pd.notnull(self._label)]
         n_samples = min(n_samples, len(samples))
 
-        np.random.seed(random_state)
+        np.random.seed = random_state
```
Comment (nirhutnik) [inline deepchecks/nlp/text_data.py:259]: nope, that is not the correct way, seed is a function...

File: tests/nlp/checks/model_evaluation/train_test_prediction_drift_test.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    # Assert
+    assert_that(condition_result, has_items(
+        equal_condition_result(is_pass=False,
+                               details="Found model prediction Cramer's V drift score of 0.04",
+                               name='categorical drift score < 0.01 and numerical drift score < 0.075')
+    ))
+
+    assert_that(result.value['Drift score'], close_to(0.04, 0.01))
+
+# TODO: Bug in combination of sampling mechanism and dummy model
```
Comment (nirhutnik) [inline tests/nlp/checks/model_evaluation/train_test_prediction_drift_test.py:39]: ?

File: tests/nlp/checks/model_evaluation/weak_segment_performance_test.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+"""Test for the nlp WeakSegmentsPerformance check"""
+
+from hamcrest import assert_that, close_to, equal_to, has_items
+
+from deepchecks.nlp.checks import WeakSegmentsPerformance
+from deepchecks.nlp.datasets.classification import tweet_emotion
+from tests.base.utils import equal_condition_result
+
+
+def test_tweet_emotion(tweet_emotion_train_test_textdata):
```
Comment (nirhutnik) [inline tests/nlp/checks/model_evaluation/weak_segment_performance_test.py:20]: Add TODO for more tests?

File: deepchecks/nlp/datasets/classification/tweet_emotion.py (hunk 3 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/datasets/classification/tweet_emotion.py:34]: Agreed

File: tests/nlp/checks/model_evaluation/train_test_prediction_drift_test.py (hunk 5 above)
Comment (Nadav-Barak) [reply tests/nlp/checks/model_evaluation/train_test_prediction_drift_test.py:39]: Opened a bug in linear
